mlflow/gateway/providers/palm.py

```diff
@@ -0,0 +1,189 @@
+from typing import Any, Dict
+
+from fastapi import HTTPException
+from fastapi.encoders import jsonable_encoder
+
+from mlflow.gateway.config import PaLMConfig, RouteConfig
+from mlflow.gateway.providers.base import BaseProvider
+from mlflow.gateway.providers.utils import rename_payload_keys, send_request
+from mlflow.gateway.schemas import chat, completions, embeddings
+
+
+class PaLMProvider(BaseProvider):
+    def __init__(self, config: RouteConfig) -> None:
+        super().__init__(config)
+        if config.model.config is None or not isinstance(config.model.config, PaLMConfig):
+            raise TypeError(f"Unexpected config type {config.model.config}")
+        self.palm_config: PaLMConfig = config.model.config
+
+    async def _request(self, path: str, payload: Dict[str, Any]) -> Dict[str, Any]:
+        return await send_request(
+            headers={},
+            base_url="https://generativelanguage.googleapis.com/v1beta3/models/",
+            path=path + f"?key={self.palm_config.palm_api_key}",
```

Comment:
/ai are there any risks with including an API key in the URL?

Reply:
Yeah unfortunately they don't seem to accept an Auth header, and others are forced to do the same: https://github.com/usagepanda/proxy/blob/main/converters/openai_to_palm.js#L28.

Reply:
Does `x-goog-api-key` work? https://developers.generativeai.google/tutorials/setup

Reply:
It does actually! Great find, updated to use that instead